docs/tutorials/time-series.md

```diff
@@ -0,0 +1,766 @@
+    same outputs.
+
+**Feldera optimizes time series processing**.  While Feldera offers the same
+expressive power for time series and non-time-series data, it can evaluate some
+classes of time series queries faster and/or using less memory:
+
+  * It takes advantage of the fact that timestamps in a time series grow monotonically or
+    almost monotonically.  This allows discarding old data no longer needed to compute outputs
+    for new data points. In order to apply this optimization the query engine needs
+    to know which columns in the input time series contain (almost) monotonically growing
+    timestamp values.  This requires
+    [`LATENESS` annotations](#timestamp-columns-and-lateness), described below.
+
+  * Time series data is often append-only: once an event has been appended to the
+    series, it cannot be deleted or modified.  Feldera implements an optimized version
+    of several operators for append-only collections.  In order to enable these
+    optimizations, the query engine needs to know which inputs are append-only.  The user
+    can provide this information to the engine using [`append_only` annotations](#append-only-tables).
+
+As a user, you do not need to understand the intricate details of time series
+processing in order to take advantage of these optimizations. Your simply
+communicate information about the input data to the query engine using
+annotations.  Feldera uses this information to apply optimizations automatically
+in a way that does not affect the output of the queries.  **Assuming that user
+annotations are correct, the optimized query plan produces the same outputs as
+the unoptimized plan.**
+
+**Users can control the timing of the output of time series queries**.  Users
+can further take advantage of the monotonicity of time series inputs to control
+**when** the output of a query is produced by Feldera using
+[`emit_final`](#emitting-final-values-of-a-view-with-emit_final)
+annotations.
+
+In the rest of this guide, we discuss these concepts in more detail.
+
+## Timestamp columns and lateness
+
+A timestamp column is a column in a SQL table or view that indicates a specific
+point in time associated with each row.
+
+  * There can be any number of timestamp columns in a table. For instance, a
+    table that describes taxi rides can store the time when the ride order was
+    received, pickup time, dropoff time, and the time when the payment was
+    processed.
+
+  * Timestamp columns can have any temporal (`TIMESTAMP`, `DATE`, `TIME`) or
+    numeric type.
+
+**A timestamp column can grow monotonically or almost monotonically**. Time
+series data often arrives ordered by timestamp, i.e., every event has the same
+or larger timestamp than the previous event.  In some cases, events can get
+reordered and delayed, but this delay is bounded, e.g., it may not exceed 1
+hour.  We refer to this bound as **lateness**.
+
+**Lateness** is a constant bound `L` associated with a timestamp column in a
```

Comment:
The more I read, the more deeply technical it becomes. The preamble is the right place in that case to instead summarize about these concepts in simpler, user-relevant terms.  E.g., it's missing something like this.

------

At a high-level, Feldera allows users to efficiently compute on timeseries data by simply providing  "lateness" annotations on tables or view columns. This annotation is a property of the data source and tells Feldera the maximum out-of-orderness in the data. For example, it allows users to convey to Feldera a hint of the form 'I know my IoT sensor will never send data that's older than a day'. Feldera can then take these hints and automatically decide when it is safe to drop state that won't affect the outcome of a view. This allows Feldera to potentially compute on infinite streams of data (like telemetry data) with finite memory, making it extremely resource efficient.

In the following sections, we will explain lateness and other key concepts with which users can efficiently compute over timeseries data using Feldera.